### DISCUSSION TITLE
Build with watch mode

### DISCUSSION THREAD

**Discussion by MomenYasser** [quasarframework/quasar#10180] [Ideas / Proposals]
I need urgently solution to enable build with watch mode like vue-cli here [https://github.com/vuejs/vue-cli/issues/1317](https://github.com/vuejs/vue-cli/issues/1317)

I need this for using quasar inside micro-frontend archticture witch requires always dist folder to be served over .net app.

So I need to watch and rebuild the dist folder after every change

**Comment by ontwikkelfabriek** [comment]
@MomenYasser 

> So I need to watch and rebuild the dist folder after every change

Sorry but what do you 'watch' for?  The 'dist' folder ? 

> after every change

after what changes?

**Reply by MomenYasser** [reply]
Of course no, I mean watch source files and rebuild after every change, Just like vue cli here [https://github.com/vuejs/vue-cli/issues/1317](https://github.com/vuejs/vue-cli/issues/1317) Please check it

**Comment by hawkeye64** [comment]
@MomenYasser There is a known issue with vue-loader and hmr using webpack 5, at the moment: https://github.com/vuejs/vue-loader/issues/1795

**Reply by ontwikkelfabriek** [reply]
@MomenYasser 

You could always run a (exernal) file diff wacher tool, that will run the 'quasar build' command...

**Comment by skanti** [comment]
There is value for such a feature. For example using `Apache` to serve the contents of `dist` and having `--watch` for quick & hot fixes. The current solution would be to recompile  and wait (10-20s) for each edit.

Hobby website developers often just have single server where testing & production sits. Hence `--watch` would allow quick production changes quick & easy.
